File: packages/text-recognition/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/textrecognition/classes/options/ProcessImageOptions.java
package io.capawesome.capacitorjs.plugins.mlkit.textrecognition.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.mlkit.textrecognition.TextRecognitionPlugin;

public class ProcessImageOptions {

    @NonNull
    private final String path;

    @NonNull
    private final String script;

    public ProcessImageOptions(@NonNull PluginCall call) throws Exception {
        String path = call.getString("path", null);
        if (path == null) {
            throw new Exception(TextRecognitionPlugin.ERROR_PATH_MISSING);
        }
        this.path = path;
        this.script = ProcessImageOptions.getScriptFromCall(call);
    }

    @NonNull
    public String getPath() {
        return path;
    }

    @NonNull
    public String getScript() {
        return script;
    }

    @NonNull
    private static String getScriptFromCall(@NonNull PluginCall call) throws Exception {
        String script = call.getString("script", "LATIN");
        switch (script) {
            case "LATIN":
            case "CHINESE":
            case "DEVANAGARI":
            case "JAPANESE":
            case "KOREAN":
                return script;
            default:
                throw new Exception(TextRecognitionPlugin.ERROR_SCRIPT_INVALID);
        }
    }
}

File: packages/text-recognition/ios/Plugin/Classes/Options/ProcessImageOptions.swift
import Foundation
import Capacitor

@objc public class ProcessImageOptions: NSObject {
    private var path: String
    private var script: String

    init(_ call: CAPPluginCall) throws {
        guard let path = call.getString("path") else {
            throw CustomError.pathMissing
        }
        self.path = path
        self.script = try ProcessImageOptions.getScriptFromCall(call)
    }

    func getPath() -> String {
        return path
    }

    func getScript() -> String {
        return script
    }

    private static func getScriptFromCall(_ call: CAPPluginCall) throws -> String {
        let script = call.getString("script", "LATIN")
        switch script {
        case "LATIN", "CHINESE", "DEVANAGARI", "JAPANESE", "KOREAN":
            return script
        default:
            throw CustomError.scriptInvalid
        }
    }
}
